pallets/gear/src/lib.rs

```diff
@@ -272,7 +272,7 @@ pub mod pallet {
             .flatten()
         }
 
-        pub fn get_gas_spent(dest: H256, payload: Vec<u8>) -> Option<u64> {
+        pub fn get_gas_spent(source: H256, dest: H256, payload: Vec<u8>) -> Option<u64> {
```

Comment:
What about making the new argument 'Option<H256>' and random it or hardcode for 'None' case?